trpcultivate_phenotypes/src/TripalCultivateValidator/ValidatorTraits/FileTypes.php

```diff
@@ -47,15 +47,17 @@ public function setFileMimeType(string $mime_type) {
       throw new \Exception("The setFileMimeType() setter requires a string of the input file's mime-type and must not be empty.");
     }
 
-    /** @TODO: Log the below message instead of throwing an exception
-     *         This check should occur in the DataFile validator
+    // Check if mime-type is in our mapping array
     if (!isset(self::$mime_to_delimiter_mapping[ $mime_type ])) {
-      throw new \Exception('The FileTypes Trait requires a supported mime-type but ' . $mime_type . ' is unsupported.');
+      // Since this is checking a user-provided value, the error is going to be
+      // logged and then checked by a validator so that the error can be passed
+      // to the user in a friendly way.
+      $this->logger->error("The setFileMimeType() setter requires a supported mime-type but '" . $mime_type . "' is unsupported.");
```

Comment:
```suggestion
      $this->logger->error("The setFileMimeType() setter requires a supported mime-type but '$mime_type' is unsupported.");
```
The variable in this message can be translated without the dot operator (concatenate).